probe-rs/src/core/mod.rs

```diff
@@ -77,6 +105,55 @@ pub(crate) enum RegisterKind {
     PC,
 }
 
+/// A value of a core register
+///
+/// Creating a new `RegisterValue` should be done using From or Into.
+/// Converting a value back to a primitive type can be done with either
+/// a match arm or TryInto
+#[derive(Debug, Clone, PartialEq)]
+pub enum RegisterValue {
+    /// 32-bit unsigned integer
+    U32(u32),
+    /// 64-bit unsigned integer
+    U64(u64),
+}
+
+impl From<u32> for RegisterValue {
+    fn from(val: u32) -> Self {
+        Self::U32(val)
+    }
+}
+
+impl From<u64> for RegisterValue {
+    fn from(val: u64) -> Self {
+        Self::U64(val)
+    }
+}
+
+impl TryInto<u32> for RegisterValue {
+    type Error = crate::Error;
+
+    fn try_into(self) -> Result<u32, Self::Error> {
+        match self {
+            Self::U32(v) => Ok(v),
+            Self::U64(v) => v
+                .try_into()
+                .map_err(|_| crate::Error::Other(anyhow!("Value too large for u32"))),
+        }
+    }
+}
+
+impl TryInto<u64> for RegisterValue {
```

Comment:
I think for this we should actually provide an `Into` impl.

Reply:
The reason it's TryInto is that before 64-bit ARM support is done I'm expecting to have to add support for a 128 bit register, so much like the u32 conversion it's not guaranteed to be safe for all registers.

Reply:
I thought about this some more and I think adding `Into` will require a break in a few more commits.  ARMv8-A has 64-bit floating point registers (rust type of f64) and 128-bit SIMD registers (rust type of u128) that would get added as arms to this TryInto match block.  Given that my view is that we should keep this as is, but please let me know if you still disagree.